src/meta/src/stream/fragmenter.rs

```diff
@@ -279,7 +282,10 @@ where
         // For HashJoin nodes, attempting to rewrite to delta joins only on inner join
         // with only equal conditions
         if let Node::HashJoinNode(hash_join_node) = stream_node.get_node()? {
-            current_fragment.table_ids_cnt += 2;
+            self.table_ids_map
```

Comment:
should insert 2 table ids?

Also please modify the comments. This part is intended to do rewrite. We now add more functionalities.

Reply:
See the comments of `table_ids_map`. HashJoin only records the left_table_id. Otherwise we use vector[u32]. I'm not sure which better?

Yes will add comments! Thanks

Reply:
I think we can directly replace the internal table id inside the join node?

Reply:
Otherwise it will be hard for delta join to get the table id...

Reply:
> I think we can directly replace the internal table id inside the join node?

Seems doable and better. But I guess delta join can also easily access the hash map -- It's in fragmenter and delta join can mutate it.

Reply:
So directly replace table id seems better.. We do not need this hashmap. I'll try.

Reply:
Yes, we may need a two phase rewrite. In this phase, assign local id. Then later, we iterate the graph again to assign global id.

Reply:
A slightly downside will be use TableRefId (proto def of table id) again in the code. But we will remove it finally. I'll consider to change TableRefId to simply u32.